--- a/kernel/nvidia-drm/nvidia-drm-encoder.c
+++ b/kernel/nvidia-drm/nvidia-drm-encoder.c
@@ -35,6 +35,8 @@
 #include <drm/drm_atomic.h>
 #include <drm/drm_atomic_helper.h>
 
+#include <linux/version.h>
+
 static void nv_drm_encoder_destroy(struct drm_encoder *encoder)
 {
     struct nv_drm_encoder *nv_encoder = to_nv_encoder(encoder);
@@ -215,8 +217,11 @@
     }
 
     /* Attach encoder and connector */
-
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 0)
+    ret = drm_connector_attach_encoder(connector, encoder);
+#else
     ret = drm_mode_connector_attach_encoder(connector, encoder);
+#endif
 
     if (ret != 0) {
         NV_DRM_DEV_LOG_ERR(

--- a/kernel/nvidia-drm/nvidia-drm-connector.c
+++ b/kernel/nvidia-drm/nvidia-drm-connector.c
@@ -35,6 +35,8 @@
 #include <drm/drm_atomic.h>
 #include <drm/drm_atomic_helper.h>
 
+#include <linux/version.h>
+
 static void nv_drm_connector_destroy(struct drm_connector *connector)
 {
     struct nv_drm_connector *nv_connector = to_nv_connector(connector);
@@ -226,7 +228,11 @@
 
 
     if (nv_connector->edid != NULL) {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 19, 0)
+        drm_connector_update_edid_property(
+#else
         drm_mode_connector_update_edid_property(
+#endif
             connector, nv_connector->edid);
     }
 

