--- a/kernel/common/inc/nv-linux.h
+++ b/kernel/common/inc/nv-linux.h
@@ -281,7 +281,8 @@ NV_STATUS nvos_forward_error_to_cray(str
  * NVIDIA graphics driver has no way of knowing and is unlikely
  * to work correctly.
  */
-#if defined(CONFIG_XEN) && !defined(CONFIG_PARAVIRT)
+#if defined(CONFIG_XEN) && !defined(CONFIG_PARAVIRT) && \
+    !defined(NVCPU_ARM)
 #include <asm/maddr.h>
 #include <xen/interface/memory.h>
 #define NV_XEN_SUPPORT_FULLY_VIRTUALIZED_KERNEL
