From ed6fe9cb91526948f3de903d8ad866f2fda4ef91 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nicolas=20Vi=C3=A9ville?= <nicolas.vieville@uphf.fr>
Date: Mon, 8 Apr 2024 17:48:38 +0200
Subject: [PATCH] gcc14: no previous prototype for nv_load_dma_map_scatterlist
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Nicolas Viéville <nicolas.vieville@uphf.fr>
---
 nvidia/nv-dma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nvidia/nv-dma.c b/nvidia/nv-dma.c
index a06c12c..343e878 100644
--- a/nvidia/nv-dma.c
+++ b/nvidia/nv-dma.c
@@ -221,7 +221,7 @@ void nv_destroy_dma_map_scatterlist(nv_dma_map_t *dma_map)
     os_free_mem(dma_map->mapping.discontig.submaps);
 }
 
-void nv_load_dma_map_scatterlist(
+static void nv_load_dma_map_scatterlist(
     nv_dma_map_t *dma_map,
     NvU64 *va_array
 )
-- 
2.44.0

