From 13c6899a34e153e498a6b1f315d72877908282fc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nicolas=20Vi=C3=A9ville?= <nicolas.vieville@uphf.fr>
Date: Sun, 10 May 2026 16:29:26 +0200
Subject: [PATCH 1/6] nv-memdbg.h: fix warnimg -Wempty-body
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Nicolas Viéville <nicolas.vieville@uphf.fr>
---
 kernel/common/inc/nv-memdbg.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/kernel/common/inc/nv-memdbg.h b/kernel/common/inc/nv-memdbg.h
index c618ead..04bf6ac 100644
--- a/kernel/common/inc/nv-memdbg.h
+++ b/kernel/common/inc/nv-memdbg.h
@@ -28,8 +28,9 @@ void nv_memdbg_exit(void);
 
 #else
 
-#define NV_MEMDBG_ADD(ptr, size)
-#define NV_MEMDBG_REMOVE(ptr, size)
+// Fix Wempty-body warnings: adding while(0)
+#define NV_MEMDBG_ADD(ptr, size) while(0)
+#define NV_MEMDBG_REMOVE(ptr, size) while(0)
 
 #endif /* NV_MEM_LOGGER */
 
-- 
2.54.0


From 7958f674b57acbd293e8d3843e93dc0007738b4a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nicolas=20Vi=C3=A9ville?= <nicolas.vieville@uphf.fr>
Date: Sun, 10 May 2026 16:51:47 +0200
Subject: [PATCH 2/6] Linux 6.19: conftest.sh: Kbuild: enable -fms-extensions
 in CFLAGS
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Nicolas Viéville <nicolas.vieville@uphf.fr>
---
 kernel/conftest.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/kernel/conftest.sh b/kernel/conftest.sh
index 8dab932..36aa4ce 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -258,6 +258,11 @@ build_cflags() {
             CFLAGS="$CFLAGS -mfentry -DCC_USING_FENTRY"
         fi
     fi
+
+    # Related to commit "Kbuild: enable -fms-extensions" (Rasmus Villemoes, 20 Oct 2025)
+    # Enable the flags since the Linux headers use those extensions in some structs
+    # See https://www.phoronix.com/news/Linux-6.19-Patch-Would-MS-Ext
+    CFLAGS="$CFLAGS -fms-extensions"
 }
 
 CONFTEST_PREAMBLE="#include \"conftest/headers.h\"
-- 
2.54.0


From 1490a3963f2b4d1a5a40dc100312fadabfb05ecf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nicolas=20Vi=C3=A9ville?= <nicolas.vieville@uphf.fr>
Date: Sun, 10 May 2026 17:18:16 +0200
Subject: [PATCH 3/6] Linux 6.19: nv-dma.c: replace map_resource with map_phys,
 related to commit "dma-mapping: remove unused mapping resource callbacks"
 (Leon Romanovsky, 15 Oct 2025)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Nicolas Viéville <nicolas.vieville@uphf.fr>
---
 kernel/nvidia/nv-dma.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/kernel/nvidia/nv-dma.c b/kernel/nvidia/nv-dma.c
index 343e878..4b942ed 100644
--- a/kernel/nvidia/nv-dma.c
+++ b/kernel/nvidia/nv-dma.c
@@ -13,6 +13,7 @@
 
 #include "os-interface.h"
 #include "nv-linux.h"
+#include <linux/version.h>
 
 NV_STATUS   nv_create_dma_map_scatterlist (nv_dma_map_t *dma_map);
 void        nv_destroy_dma_map_scatterlist(nv_dma_map_t *dma_map);
@@ -619,7 +620,13 @@ static NvBool nv_dma_is_map_resource_implemented
 #endif
     }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+    // Rel. commit "dma-mapping: remove unused mapping resource callbacks" (Leon Romanovsky, 15 Oct 2025)
+    // https://lore.kernel.org/all/20251015-remove-map-page-v5-0-3bbfe3a25cdf@kernel.org/
+    return (ops->map_phys != NULL);
+#else
     return (ops->map_resource != NULL);
+#endif
 #else
     return NV_FALSE;
 #endif
-- 
2.54.0


From 56193b13dba70224898877766d97962eba536b68 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nicolas=20Vi=C3=A9ville?= <nicolas.vieville@uphf.fr>
Date: Sun, 10 May 2026 17:37:44 +0200
Subject: [PATCH 4/6] Linux 6.19: nvlink_linux.c, os-interface.c: replace
 in_irq with in_hardirq added in v5.11-rc1 (2020-12-15)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Nicolas Viéville <nicolas.vieville@uphf.fr>
---
 kernel/nvidia/nvlink_linux.c |  9 +++++++++
 kernel/nvidia/os-interface.c | 17 +++++++++++++++++
 2 files changed, 26 insertions(+)

diff --git a/kernel/nvidia/nvlink_linux.c b/kernel/nvidia/nvlink_linux.c
index 9f7d8c1..723ee8e 100644
--- a/kernel/nvidia/nvlink_linux.c
+++ b/kernel/nvidia/nvlink_linux.c
@@ -31,6 +31,11 @@
 #include "nv-procfs.h"
 #include "nv-time.h"
 #include "nvlink_proto.h"
+#include <linux/version.h>
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+#include <linux/hardirq.h>
+#endif
 
 #define MAX_ERROR_STRING           512
 
@@ -571,7 +576,11 @@ void NVLINK_API_CALL nvlink_sleep(unsigned int ms)
 
     nv_gettimeofday(&tm_aux);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+    if (in_hardirq() && (ms > NV_MAX_ISR_DELAY_MS))
+#else
     if (in_irq() && (ms > NV_MAX_ISR_DELAY_MS))
+#endif
     {
         return;
     }
diff --git a/kernel/nvidia/os-interface.c b/kernel/nvidia/os-interface.c
index 262171f..c3e9da7 100644
--- a/kernel/nvidia/os-interface.c
+++ b/kernel/nvidia/os-interface.c
@@ -17,6 +17,11 @@
 #include "nv-time.h"
 #include "nv-gpu-numa.h"
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+#include <linux/hardirq.h>
+#endif
+
 #define MAX_ERROR_STRING 512
 static char nv_error_string[MAX_ERROR_STRING];
 nv_spinlock_t nv_error_string_lock;
@@ -193,7 +198,11 @@ BOOL NV_API_CALL os_semaphore_may_sleep(void)
 
 BOOL NV_API_CALL os_is_isr(void)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+    return (in_hardirq());
+#else
     return (in_irq());
+#endif
 }
 
 // return TRUE if the caller is the super-user
@@ -510,7 +519,11 @@ NV_STATUS NV_API_CALL os_delay_us(NvU32 MicroSeconds)
     nv_gettimeofday(&tm1);
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+    if (in_hardirq() && (MicroSeconds > NV_MAX_ISR_DELAY_US))
+#else
     if (in_irq() && (MicroSeconds > NV_MAX_ISR_DELAY_US))
+#endif
         return NV_ERR_GENERIC;
     
     mdelay_safe_msec = MicroSeconds / 1000;
@@ -555,7 +568,11 @@ NV_STATUS NV_API_CALL os_delay(NvU32 MilliSeconds)
     tm_start = tm_aux;
 #endif
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+    if (in_hardirq() && (MilliSeconds > NV_MAX_ISR_DELAY_MS))
+#else
     if (in_irq() && (MilliSeconds > NV_MAX_ISR_DELAY_MS))
+#endif
         return NV_ERR_GENERIC;
 
     if (!NV_MAY_SLEEP()) 
-- 
2.54.0


From f02cfacbf9a245b351b42bd71e8cc05d128d9cf6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nicolas=20Vi=C3=A9ville?= <nicolas.vieville@uphf.fr>
Date: Sun, 10 May 2026 17:43:12 +0200
Subject: [PATCH 5/6] Linux 6.19: nvidia-drm-priv.h: add drm_print.h, Related
 to commit "drm/mm: replace drm_print.h include with a forward declaration"
 (Jani Nikula, 29 Oct 2025)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Nicolas Viéville <nicolas.vieville@uphf.fr>
---
 kernel/nvidia-drm/nvidia-drm-priv.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kernel/nvidia-drm/nvidia-drm-priv.h b/kernel/nvidia-drm/nvidia-drm-priv.h
index 4f1a0e7..953c2eb 100644
--- a/kernel/nvidia-drm/nvidia-drm-priv.h
+++ b/kernel/nvidia-drm/nvidia-drm-priv.h
@@ -39,6 +39,9 @@
 #include <drm/drm_gem.h>
 #endif
 
+// Rel. commit "drm/mm: replace drm_print.h include with a forward declaration" (Jani Nikula, 29 Oct 2025)
+#include <drm/drm_print.h>
+
 #include "nvidia-drm-os-interface.h"
 
 #include "nvkms-kapi.h"
-- 
2.54.0


From abd458efef4cf38c2d98e9562f92a37048e90555 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nicolas=20Vi=C3=A9ville?= <nicolas.vieville@uphf.fr>
Date: Sun, 10 May 2026 18:18:49 +0200
Subject: [PATCH 6/6] Linux 6.19: nvidia-drm-drv.c, nvidia-drm-helper.c,
 nvidia-drm-helper.h, nvidia-drm-modeset.c: replace nv_drm_for_each_* with
 for_each_*
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Nicolas Viéville <nicolas.vieville@uphf.fr>
---
 kernel/nvidia-drm/nvidia-drm-crtc.c    | 11 +++++++++++
 kernel/nvidia-drm/nvidia-drm-drv.c     |  8 ++++++++
 kernel/nvidia-drm/nvidia-drm-helper.c  |  3 +++
 kernel/nvidia-drm/nvidia-drm-helper.h  |  3 +++
 kernel/nvidia-drm/nvidia-drm-modeset.c | 23 +++++++++++++++++++++++
 5 files changed, 48 insertions(+)

diff --git a/kernel/nvidia-drm/nvidia-drm-crtc.c b/kernel/nvidia-drm/nvidia-drm-crtc.c
index 6d5478e..a099cb1 100644
--- a/kernel/nvidia-drm/nvidia-drm-crtc.c
+++ b/kernel/nvidia-drm/nvidia-drm-crtc.c
@@ -39,6 +39,8 @@
 #include <drm/drm_atomic.h>
 #include <drm/drm_atomic_helper.h>
 
+#include <linux/version.h>
+
 static const u32 nv_default_supported_plane_drm_formats[] = {
     DRM_FORMAT_ARGB1555,
     DRM_FORMAT_XRGB1555,
@@ -151,7 +153,11 @@ static int nv_drm_plane_atomic_check(struct drm_plane *plane,
         goto done;
     }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+    for_each_new_crtc_in_state(plane_state->state, crtc, crtc_state, i) {
+#else
     nv_drm_for_each_crtc_in_state(plane_state->state, crtc, crtc_state, i) {
+#endif
         struct nv_drm_crtc_state *nv_crtc_state = to_nv_crtc_state(crtc_state);
         struct NvKmsKapiHeadRequestedConfig *head_req_config =
             &nv_crtc_state->req_config;
@@ -368,8 +374,13 @@ static int nv_drm_crtc_atomic_check(struct drm_crtc *crtc,
 
         req_config->flags.displaysChanged = NV_TRUE;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+        for_each_new_connector_in_state(crtc_state->state,
+                                        connector, connector_state, j) {
+#else
         nv_drm_for_each_connector_in_state(crtc_state->state,
                                            connector, connector_state, j) {
+#endif
             if (connector_state->crtc != crtc) {
                 continue;
             }
diff --git a/kernel/nvidia-drm/nvidia-drm-drv.c b/kernel/nvidia-drm/nvidia-drm-drv.c
index cd0bcac..3183ff7 100644
--- a/kernel/nvidia-drm/nvidia-drm-drv.c
+++ b/kernel/nvidia-drm/nvidia-drm-drv.c
@@ -558,13 +558,21 @@ void nv_drm_master_drop(struct drm_device *dev, struct drm_file *file_priv)
 
     drm_modeset_lock_all(dev);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+    if ((err = drm_atomic_helper_disable_all(
+#else
     if ((err = nv_drm_atomic_helper_disable_all(
+#endif
             dev,
             dev->mode_config.acquire_ctx)) != 0) {
 
         NV_DRM_DEV_LOG_ERR(
             nv_dev,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+            "drm_atomic_helper_disable_all failed with error code %d !",
+#else
             "nv_drm_atomic_helper_disable_all failed with error code %d !",
+#endif
             err);
     }
 
diff --git a/kernel/nvidia-drm/nvidia-drm-helper.c b/kernel/nvidia-drm/nvidia-drm-helper.c
index 8b89afe..5b4ed47 100644
--- a/kernel/nvidia-drm/nvidia-drm-helper.c
+++ b/kernel/nvidia-drm/nvidia-drm-helper.c
@@ -28,6 +28,7 @@
  */
 
 #include "nvidia-drm-helper.h"
+#include <linux/version.h>
 
 #if defined(NV_DRM_ATOMIC_MODESET_AVAILABLE)
 
@@ -62,6 +63,7 @@ static void __nv_drm_framebuffer_put(struct drm_framebuffer *fb)
 
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 19, 0)
 /*
  * drm_atomic_helper_disable_all() has been added by commit
  * 1494276000db789c6d2acd85747be4707051c801, which is Signed-off-by:
@@ -199,4 +201,5 @@ free:
     return ret;
 }
 
+#endif /* kernel < 6.19 */
 #endif /* NV_DRM_ATOMIC_MODESET_AVAILABLE */
diff --git a/kernel/nvidia-drm/nvidia-drm-helper.h b/kernel/nvidia-drm/nvidia-drm-helper.h
index 831737d..bc61d7a 100644
--- a/kernel/nvidia-drm/nvidia-drm-helper.h
+++ b/kernel/nvidia-drm/nvidia-drm-helper.h
@@ -24,6 +24,7 @@
 #define __NVIDIA_DRM_HELPER_H__
 
 #include "nvidia-drm-conftest.h"
+#include <linux/version.h>
 
 #if defined(NV_DRM_AVAILABLE)
 
@@ -156,6 +157,7 @@ nv_drm_prime_pages_to_sg(struct drm_device *dev,
 #include <drm/drm_atomic.h>
 #include <drm/drm_atomic_helper.h>
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 19, 0)
 int nv_drm_atomic_helper_disable_all(struct drm_device *dev,
                                      struct drm_modeset_acquire_ctx *ctx);
 
@@ -277,6 +279,7 @@ int nv_drm_atomic_helper_disable_all(struct drm_device *dev,
     for_each_plane_in_state(__state, plane, plane_state, __i)
 #endif
 
+#endif /* Kernel < 6.19 */
 static inline struct drm_crtc *nv_drm_crtc_find(struct drm_device *dev,
     uint32_t id)
 {
diff --git a/kernel/nvidia-drm/nvidia-drm-modeset.c b/kernel/nvidia-drm/nvidia-drm-modeset.c
index 3a57e8e..b6ef1e7 100644
--- a/kernel/nvidia-drm/nvidia-drm-modeset.c
+++ b/kernel/nvidia-drm/nvidia-drm-modeset.c
@@ -21,6 +21,7 @@
  */
 
 #include "nvidia-drm-conftest.h" /* NV_DRM_ATOMIC_MODESET_AVAILABLE */
+#include <linux/version.h>
 
 #if defined(NV_DRM_ATOMIC_MODESET_AVAILABLE)
 
@@ -112,12 +113,19 @@ nv_drm_atomic_apply_modeset_config(struct drm_device *dev,
     struct NvKmsKapiRequestedModeSetConfig *requested_config =
         &(to_nv_atomic_state(state)->config);
     struct drm_crtc *crtc;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+    struct drm_crtc_state *old_crtc_state, *new_crtc_state;
+#else
     struct drm_crtc_state *crtc_state;
+#endif
     int i;
 
     memset(requested_config, 0, sizeof(*requested_config));
 
     /* Loop over affected crtcs and construct NvKmsKapiRequestedModeSetConfig */
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+    for_each_oldnew_crtc_in_state(state, crtc, old_crtc_state, new_crtc_state, i) {
+#else
     nv_drm_for_each_crtc_in_state(state, crtc, crtc_state, i) {
         /*
          * When commiting a state, the new state is already stored in
@@ -126,6 +134,7 @@ nv_drm_atomic_apply_modeset_config(struct drm_device *dev,
          */
         struct drm_crtc_state *new_crtc_state =
                                commit ? crtc->state : crtc_state;
+#endif
         struct nv_drm_crtc *nv_crtc = to_nv_crtc(crtc);
 
         requested_config->headRequestedConfig[nv_crtc->head] =
@@ -134,7 +143,9 @@ nv_drm_atomic_apply_modeset_config(struct drm_device *dev,
         requested_config->headsMask |= 1 << nv_crtc->head;
 
         if (commit) {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 19, 0)
             struct drm_crtc_state *old_crtc_state = crtc_state;
+#endif
             struct nv_drm_crtc_state *nv_new_crtc_state =
                 to_nv_crtc_state(new_crtc_state);
 
@@ -220,7 +231,11 @@ int nv_drm_atomic_commit(struct drm_device *dev,
 
     int i;
     struct drm_crtc *crtc = NULL;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+    struct drm_crtc_state *new_crtc_state = NULL;
+#else
     struct drm_crtc_state *crtc_state = NULL;
+#endif
     struct nv_drm_device *nv_dev = to_nv_device(dev);
 
     /*
@@ -230,7 +245,11 @@ int nv_drm_atomic_commit(struct drm_device *dev,
      * updates to complete.
      */
     if (nonblock) {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+        for_each_new_crtc_in_state(state, crtc, new_crtc_state, i) {
+#else
         nv_drm_for_each_crtc_in_state(state, crtc, crtc_state, i) {
+#endif
             struct nv_drm_crtc *nv_crtc = to_nv_crtc(crtc);
 
             /*
@@ -303,7 +322,11 @@ int nv_drm_atomic_commit(struct drm_device *dev,
     }
 
     if (ret == 0 && !nonblock) {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+        for_each_new_crtc_in_state(state, crtc, new_crtc_state, i) {
+#else
         nv_drm_for_each_crtc_in_state(state, crtc, crtc_state, i) {
+#endif
             struct nv_drm_crtc *nv_crtc = to_nv_crtc(crtc);
 
             /*
-- 
2.54.0

