diff --git a/nvidia/nv-gpu-numa.c b/nvidia/nv-gpu-numa.c
index 7916d35..646cc0d 100644
--- a/nvidia/nv-gpu-numa.c
+++ b/nvidia/nv-gpu-numa.c
@@ -177,6 +177,19 @@ static int filldir_get_memblock_id(struct dir_context *ctx,
     return 0;
 }
 
+// Wrapper function to cast between incompatible function types from: 
+// int (*)(struct dir_context *, const char *, int,  loff_t,  u64,  unsigned int) to 
+// bool (*)(struct dir_context *, const char *, int,  loff_t,  u64,  unsigned int)
+static bool b_filldir_get_memblock_id(struct dir_context *ctx,
+                                   const char *name,
+                                   int name_len,
+                                   loff_t offset,
+                                   u64 ino,
+                                   unsigned int d_type)
+{
+    return filldir_get_memblock_id(ctx, name, name_len, offset, ino, d_type);
+}
+ 
 /*
  * Brings memory block online using the sysfs memory-hotplug interface
  *   https://www.kernel.org/doc/Documentation/memory-hotplug.txt
@@ -229,7 +242,7 @@ static NV_STATUS gather_memblock_ids_for_node
     char numa_file_path[BUF_SIZE];
     struct file *filp;
     int err; 
-    nv_dir_context_t ats_ctx = { .ctx.actor = (filldir_t)filldir_get_memblock_id };
+    nv_dir_context_t ats_ctx = { .ctx.actor = (filldir_t)b_filldir_get_memblock_id };
 
     memset(numa_file_path, 0, sizeof(numa_file_path));
     sprintf(numa_file_path, "%s%d", NID_PATH, node_id);
