diff --git a/conftest.sh b/conftest.sh
index beaae93..386faf4 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -3312,16 +3312,17 @@ compile_test() {
             # Determine if the pnv_npu2_init_context() function is
             # present.
             #
-            CODE="
+            #CODE="
             #if defined(NV_ASM_POWERNV_H_PRESENT)
             #include <linux/pci.h>
             #include <asm/powernv.h>
             #endif
-            void conftest_pnv_npu2_init_context(void) {
-                pnv_npu2_init_context();
-            }"
+            #void conftest_pnv_npu2_init_context(void) {
+            #    pnv_npu2_init_context();
+            #}"
 
-            compile_check_conftest "$CODE" "NV_PNV_NPU2_INIT_CONTEXT_PRESENT" "" "functions"
+            #compile_check_conftest "$CODE" "NV_PNV_NPU2_INIT_CONTEXT_PRESENT" "" "functions"
+            echo "#undef NV_PNV_NPU2_INIT_CONTEXT_PRESENT" | append_conftest "functions"
         ;;
 
         drm_driver_unload_has_int_return_type)
